index.js

```diff
@@ -454,14 +475,15 @@ PersonaClient.prototype.obtainToken = function (id, secret, callback) {
                 req.write(post_data);
                 req.end();
             } else {
-                _this.debug("Found cached token for key " + cacheKey + ": " + reply);
                 var data;
-                try {
+                if (_.isObject(reply)) {
+                    data = reply;
+                    reply = JSON.stringify(data);
```

Comment:
The problem is that it needs to be marshaled differently depending on the cache store: it might either be an object or a json string coming out of cache.